server/src/test/java/org/elasticsearch/search/query/PartialHitCountCollectorTests.java

```diff
@@ -101,15 +107,33 @@ public void testCollectedHitCount() throws Exception {
     }
 
     public void testCollectedHitCountEarlyTerminated() throws Exception {
-        Query query = new BooleanQuery.Builder().add(new TermQuery(new Term("string", "a1")), BooleanClause.Occur.SHOULD)
-            .add(new TermQuery(new Term("string", "b3")), BooleanClause.Occur.SHOULD)
-            .add(new TermQuery(new Term("string", "b6")), BooleanClause.Occur.SHOULD)
-            .build();
+        Query query = new NonCountingTermQuery(new Term("string", "foo"));
         // there's three docs matching the query: any totalHitsThreshold lower than 3 will trigger early termination
         int totalHitsThreshold = randomInt(2);
         PartialHitCountCollector partialHitCountCollector = new PartialHitCountCollector(totalHitsThreshold);
         searcher.search(query, partialHitCountCollector);
+
         assertEquals(totalHitsThreshold, partialHitCountCollector.getTotalHits());
         assertTrue(partialHitCountCollector.hasEarlyTerminated());
     }
+
+    /**
+     * A {@link TermQuery} which cannot terminate collection early
+     */
+    public static class NonCountingTermQuery extends TermQuery {
```

Comment:
let's make it a top-level class if we share it between tests? maybe it can be package private though given it's only used from the query package?

Reply:
I prefer package private tbh, I was already considering copying it in both test classes given that its very specific and if you don't know it exists you will hardly be able to reuse it.

Reply:
by package private I mean top-level , but only visible within the same package. It is now an inner class, and public?

Reply:
It shouldn't be public anymore, I pushed changes yesterday. I can pull it out as well but I think it has limited value.

Reply:
I read public static class still. My comment was because you were sharing it with QueryPhaseTests. If you don't, then it is not necessary to make it top-level. If it's only used in this test, it can be private

Reply:
> I read public static class still.
Not on the last commit I pushed yesterday [3391b5f](https://github.com/elastic/elasticsearch/pull/97389/commits/3391b5f58cb1402f20b712b32445bc61fd33a245). This code snippet is outdated.